rcl/src/rcl/init.c

```diff
@@ -179,7 +179,13 @@ rcl_shutdown(rcl_context_t * context)
     return rcl_convert_rmw_ret_to_rcl_ret(rmw_ret);
   }
 
-  rcl_ret_t rcl_ret = rcl_logging_fini();
+  rcl_ret_t rcl_ret = rcl_context_fini(context);
```

Comment:
I'm not sure about this change. I believe it is not assumed that `rcl_shutdown` will take care of this memory cleanup. Other places in our code are handling the context finalization alongside `rcl_shutdown`, e.g. 

https://github.com/ros2/rcl/blob/0c4c743d29d3c313a46d1079003e49081a64e292/rcl/test/rcl/test_node.cpp#L133-L134

It also seems contrary to the documentation of `rcl_context_t`: http://docs.ros2.org/latest/api/rcl/structrcl__context__t.html

There is a valid state between shutdown and finalization.